internal/utils/flags/flags.go

```diff
@@ -243,3 +245,29 @@ func (f StringSliceFlag) Register(fs *pflag.FlagSet) {
 func MarkHidden(fs *pflag.FlagSet, name string) {
 	fs.MarkHidden(name) //nolint: errcheck
 }
+
+// Deprecator is used to mark a flag for deprecation with Message, if To is provided a forwarding message will
+// be generated and used as Message.
+type Deprecator struct {
+	FirstUnsupportedVersion string
+	To                      string
```

Comment:
my issue with this approach is I don't think we can guarantee all flags being deprecated will have both a known unsupported version AND a flag that serves as its replacement.  this messaging becomes very brittle for us

considering this is really our first forray into deprecating flags, and the fact its only in 3 different commands, keeping this just as a string value is sufficient for us.  yes, it means we duplicate the deprecation string/message in 3 spots, but that's similar to how we define the usage strings anyways

can we just have `Deprecate` as a `string` field on the `Meta`?

that way, its usage in the various `Register` methods would just look like:

```go
if f.Deprecate != "" {
  MarkDeprecated(fs, f.Name, f.Deprecate)
} else if f.Hidden {
  MarkHidden(fs, f.Name)
}
```

---

in terms of the actual message, when we're setting the `Deprecate` message on these flags' `Meta`, lets go with something like:

```go
Deprecate: fmt.Sprintf("support will be removed in v3.x, please use %q instead", flagIncludeNodeModules)
// NOTE: using '%q' here is matching the pattern in our usage strings and error messages
```